src/AzureAuth/PublicClientAuth.cs

```diff
@@ -37,26 +41,46 @@ public PublicClientAuth(ILogger logger, IEnv env, ITelemetryService telemetrySer
         }
 
         /// <inheritdoc/>
-        public TokenResult Token(Guid client, Guid tenant, IEnumerable<string> scopes, AuthMode[] authModes, string domain, string prompt, TimeSpan timeout)
+        public TokenResult Token(Guid client, Guid tenant, IEnumerable<string> scopes, IEnumerable<AuthMode> authModes, string domain, string prompt, TimeSpan timeout, EventData eventData)
         {
             var result = this.tokenFetcher.AccessToken(
                 this.logger,
                 client,
                 tenant,
                 scopes,
-                authModes.Combine().PreventInteractionIfNeeded(this.env),
+                authModes.Combine().PreventInteractionIfNeeded(this.env, this.logger),
                 domain,
                 PromptHint.Prefixed(prompt),
                 timeout);
 
+            // Report individual auth flow telemetry
             result.Attempts.SendTelemetry(this.telemetryService);
 
-            if (result.Success == null)
+            var totalErrorCount = result.Attempts.SelectMany(attempt => attempt.Errors).Count();
+            eventData.Add("error_count", totalErrorCount);
+            eventData.Add("authflow_count", result.Attempts.Count);
+
+            var authflow = result.Success;
+            if (authflow == null)
             {
+                foreach (var attempt in result.Attempts)
+                {
+                    this.logger.LogDebug($"{attempt.AuthFlowName} failed after {attempt.Duration.TotalSeconds:0.00} sec. Error count: {attempt.Errors.Count}");
+                    foreach (var e in attempt.Errors)
+                    {
+                        this.logger.LogDebug($"  {e.Message}");

```

Comment:
We should add tests to asserts these logs. :)

Reply:
Same response from the first comment, a follow up PR will contain another integration test, or 2 about error cases and what get's logged.